### PR TITLE
enable everything on stable toolchain

### PR DESCRIPTION
**Pull request by ccbrown**
I'd like to be able to use the registers with stable Rust, and as far as I can tell, there was nothing really in the library that actually required nightly.

Is it reasonable to just remove the nightly requirement like this?

### CODE REVIEW

**Comment by andre-richter** [comment]
Ah! It went past me that the `asm!` macro is now stable. Can you modify `.github/workflows/` as well so that GitHub Actions doesn't switch to a nightly compiler and builds with stable?

File: Cargo.toml [hunk 1]
```diff
@@ -20,8 +20,7 @@ exclude = [
 ]
 
 [features]
-default = ["nightly"]
-nightly = ["tock-registers"]
+default = ["tock-registers"]
```
Comment (andre-richter) [inline Cargo.toml]: We should be able to remove the whole feature gating now.

File: src/lib.rs [hunk 2]
```diff
@@ -77,10 +71,8 @@
 //! ARMv8, for ARMv8-A architecture
 //! profile](https://static.docs.arm.com/ddi0487/ca/DDI0487C_a_armv8_arm.pdf?_ga=2.266626254.1122218691.1534883460-1326731866.1530967873).
 
-#![cfg_attr(feature = "nightly", feature(core_intrinsics))]
-#![cfg_attr(feature = "nightly", feature(custom_inner_attributes))]
 #![no_std]
 
 pub mod asm;
-#[cfg(feature = "nightly")]
+#[cfg(feature = "tock-registers")]
```
Comment (andre-richter) [inline src/lib.rs]: We should be able to remove the whole feature gating now.

**Comment by ccbrown** [comment]
> Ah! It went past me that the `asm!` macro is now stable. Can you modify `.github/workflows/` as well so that GitHub Actions doesn't switch to a nightly compiler and builds with stable?

Yep, done.

**Comment by berkus** [comment]
👍🏼  beautiful

**Pull request merged**